src/solver/types.jl

```diff
@@ -427,10 +427,18 @@ the default configurations set in `default_config_dict()`.
 function AtmosConfig(config::Dict; comms_ctx = nothing)
     config = override_default_config(config)
     FT = config["FLOAT_TYPE"] == "Float64" ? Float64 : Float32
-    toml_dict = CP.create_toml_dict(
-        FT;
-        override_file = CP.merge_toml_files(config["toml"]),
-    )
+
+    # Check that config["toml"] is a filepath and the file does not exist
+    if config["toml"] isa AbstractString && !isfile(config["toml"])
+        @info "Could not find TOML file at path $(joinpath(pwd(), config["toml"]))"
+        prepend_path = joinpath(dirname(@__FILE__), "..", "..", config["toml"])
```

Comment:
I think the variable name `prepend_path` doesn't describe what the variable actually is. Maybe `toml_path_in_climaatmos_dir` or something like that?

Reply:
@charleskawczynski I don't think there's a better way of doing this in `src`. I am happy to include a commented snippet for devs copy-pasting into the REPL, something like `prepend_path = joinpath(pkgdir(CA), config["toml"])`

Reply:
IMO, that means that here is a problem with the way we are doing this. I think that goes along the lines of @simonbyrne's suggestion to avoid grabbing things outside of `src/`. Maybe this logic can be extracted and moved into the driver, and we can simplify how `AtmosConfig` works?

Reply:
I don't see how that would work - running from anywhere with any configuration/parameter files isn't really needed in the driver, which is almost always run from `pkgdir(CA)`, but is needed for `AtmosConfig`.
It seems like the only way to make this work with the current system is this partial fix which assumes that parameter paths are relative to `pkgdir(CA)`. If that is not acceptable, we will probably need to rework with how parameters are taken in.

Reply:
I do agree with @charleskawczynski that ideally stuff in `src/` should not depend on stuff outside of it.

I think that `src/` should be treated as a library, so we when we mix it with stuff outside we are entangling library with an application of it.

> It seems like the only way to make this work with the current system is this partial fix which assumes that parameter paths are relative to pkgdir(CA)

I think that this is pretty much equivalent to hardcoding the parameters from a user point of view. (The point of view I take is 'how would this work if I were to just Pkg.add("ClimaAtmos")?' )